packages/babel-core/src/config/plugin.js

```diff
@@ -2,8 +2,8 @@
 
 export default class Plugin {
   constructor(plugin: {}, key?: string) {
-    if (plugin.name != null && typeof plugin.name !== "string") {
-      throw new Error("Plugin .name must be a string, null, or undefined");
+    if (!plugin.name || typeof plugin.name !== "string") {
+      throw new Error("Plugin .name must be a valid JS identifier of type string");
```

Comment:
No, I'm commenting on your error message. the "valid JS identifier of type string" is needlessly wordy, just say "needs to be a string"